### ISSUE TITLE
Don't give away awards so easily

### ISSUE THREAD

**Issue by SilPho** [CorsixTH/CorsixTH#2230]
### What feature or enhancement would you like to add?

Some of the end-of-year awards are given away too easily, particularly in the first year of a hospital.

In the original game, if you let the year run its course without doing anything at all, the game awards no trophies or awards, good or bad. Conversely, in CorsixTH you receive the zero deaths trophy ($10,000), the cure-to-death ratio award ($4000),the "low deaths" award ($3000), the "not many cures" award (-$3000) and the "low hospital value" award (Loss of reputation). That means you earn $14,000 for doing nothing at all.

In the most extreme case, the "Avatar" scenario can be beaten by simply doing nothing for 3 in-game years to push you over the only requirement: $50,000.

**Proposal:** The cure-releated rewards should apply if at least some patients are actually cured. The low hospital value award could be kept as is since it does act as a detterant against minimal play. I'm not sure if that one was in the original game or not.

### How would this feature benefit the gameplay of CorsixTH?

It would bring the behaviour closer to the original game and prevent a slight "cheese" by simply waiting a while before opening the hospital to earn a tonne of cash.

### Was the feature in the original game?

Yes

### What version of CorsixTH are you using?

0.66

### Does this feature/enhancement relate to a current problem?

No

**Comment by lewri** [comment]
If you wanted to document all the awards behaviour we have first, it may make it easier then to look at adjustments, that's generally what I did with mugmuggy's help when reworking the VIP.

But yes, some are definitely unbalanced. Do the end of year cheat and watch the rewards roll in.

**Comment by mugmuggy** [comment]
Those pointed out are missing minimum patient/visitor checks. I made similar observation in https://github.com/CorsixTH/CorsixTH/issues/967#issuecomment-367990900

From memory the annual report with visitors/cured counts, aren't necessarily values we capture either or the awards aren't applying correctly as we don't process the values from the level configs to match so its been implemented as step to have functionality there.

**Comment by SilPho** [comment]
Good point about documenting the existing behaviour. Here it is:

CTY = Cured this year
DTY = Deaths this year
NCTY = Number of patients that were kicked, got fed up, or refused to pay. It does not include deaths.
config = The level config's value for that particular reward.

First up: Trophies (At the bottom of the award screen)

**Trophies independant of one another:**
Soda award - Check if number of sodas sold this year is over the target (as per config)
VIP visits - Check if number of pleased vips is above 0 and is the same as the number of visited vips
Repuation - Awarded if the hospital reputation was never less (or even equal) to the minimum target (as per config)

** Give the first of these trophies, if eligible **
Cured - If CTY is 2 or more and DTY is 0 and NCTY is also 0.
or
No deaths - If DTY is 0
or
Many cured - If CTY is more than 90% of NCTY.

Then, awards/certificates that appear on the sides

** Pairs of awards - Can award one from any number of these pairs **
High/Low Reputation - If the current hospital reputation is above or below the config values
High/Low Value - If the current hospital is above or below the config values
High/Low Cures - If CTY is above or below the config values

** Give the first of these certificates, if eligible **
No deaths - If CTY is 2 or more and DTY is 0 and NCTY is also 0 (Identical to the trophy)
or
Low deaths - If DTY is below the config value
or
High deaths - If DTY is above the config value

** Certificates based on the cure ratio **
High/Low Cure Ratio - If CTY / DTY is above or below the config value (Default values are 5 for high and 1 for low. Works out to above 80% or below 50% by default)

**Comment by SilPho** [comment]
However, while doing some testing on this, it looks like the values for cures and not-cures are not being reset to 0 after each year. Number of deaths is being reset though.

I'd expect Hospital:onEndYear() to reset the number of cures and not-cures, but I'm not seeing that. Maybe I'm mistaken? (Checked in 0.66 and master)

**Comment by MRMIdAS** [comment]
Indeed, the "Killed Nobody" trophy, and the "Cures To Deaths" awards should be exclusive, you get one or the other (or neither) but never both.

**Comment by lewri** [comment]
> I'd expect Hospital:onEndYear() to reset the number of cures and not-cures, but I'm not seeing that. Maybe I'm mistaken? (Checked in 0.66 and master)

Originally I thought it was controlled by these rolling stats...
https://github.com/CorsixTH/CorsixTH/blob/54c0adc37caa612bd9e154f5f56958cc87ec57e9/CorsixTH/Lua/hospital.lua#L1021-L1033

But no it's not. We're not resetting the `_ty` variables anywhere that I can see :(  - It could be related to this issue, or a separate one.

**Comment by lewri** [comment]
> > I'd expect Hospital:onEndYear() to reset the number of cures and not-cures, but I'm not seeing that. Maybe I'm mistaken? (Checked in 0.66 and master)
> 
> Originally I thought it was controlled by these rolling stats...
> 
> https://github.com/CorsixTH/CorsixTH/blob/54c0adc37caa612bd9e154f5f56958cc87ec57e9/CorsixTH/Lua/hospital.lua#L1021-L1033
> 
> But no it's not. We're not resetting the `_ty` variables anywhere that I can see :( - It could be related to this issue, or a separate one.

This is an easy fix generally... add it to `Hospital:onEndYear()` to reset to 0 (just make sure it happens after annual report).

**Comment by Alberth289346** [comment]
Why not stick updating the `_ty` variables into the above computation, so all stats variables are updated together?

No problems with order of things, and less places that touch only a part of the variable set so you have to look for other parts and figure out how they relate.
Keeping related things together makes life simpler.

**Comment by TheCycoONE** [comment]
If we have consensus on the proper behaviour now this is a good first issue.

**Comment by ARGAMX** [comment]
Probably resolved by #2806

**Comment by lewri** [comment]
Resolved by #2806
